test/proposal_settings_test.rb

```diff
@@ -0,0 +1,195 @@
+      let(:use_ssh) { true }
+
+      it "sets ssh service to be enabled" do
+        expect_any_instance_of(described_class).to receive(:enable_sshd!)
+
+        described_class.create_instance
+      end
+
+      it "sets the ssh port to be opened" do
+        expect_any_instance_of(described_class).to receive(:open_ssh!)
+
+        described_class.create_instance
+      end
+    end
+
+    context "when vnc has been enable by Linuxrc" do
+      let(:use_vnc) { true }
+
+      it "sets the vnc port to be opened" do
+        expect_any_instance_of(described_class).to receive(:open_vnc!)
+
+        described_class.create_instance
+      end
+    end
+  end
+
+  describe "#enable_firewall!" do
+    it "sets firewalld service to be enabled" do
+      allow(Yast::PackagesProposal).to receive("AddResolvables")
+        .with("firewall", :package, ["firewalld"])
+
+      subject.enable_firewall!
+      expect(subject.enable_firewall).to be(true)
+    end
+
+    it "adds the firewalld package to be installed" do
+      expect(Yast::PackagesProposal).to receive("AddResolvables")
+        .with("firewall", :package, ["firewalld"])
+
+      subject.enable_firewall!
+    end
+  end
+
+  describe "#disable_firewall!" do
+    it "sets firewalld service to be disabled" do
+      allow(Yast::PackagesProposal).to receive("RemoveResolvables")
+        .with("firewall", :package, ["firewalld"])
+
+      subject.disable_firewall!
+      expect(subject.enable_firewall).to be(false)
+    end
+
+    it "removes the firewalld package for current selection" do
+      expect(Yast::PackagesProposal).to receive("RemoveResolvables")
+        .with("firewall", :package, ["firewalld"])
+
+      subject.disable_firewall!
+    end
+  end
+
+  describe "#enable_sshd!" do
+    it "sets sshd service to be enabled" do
+      allow(Yast::PackagesProposal).to receive("AddResolvables")
+        .with("firewall", :package, ["openssh"])
+
+      subject.enable_sshd!
+      expect(subject.enable_sshd).to be(true)
+    end
+
+    it "adds the openssh package to be installed" do
+      expect(Yast::PackagesProposal).to receive("AddResolvables")
+        .with("firewall", :package, ["openssh"])
+
+      subject.enable_sshd!
+    end
+  end
+
+  describe "#disable_sshd!" do
+    it "sets sshd service to be disabled" do
+      allow(Yast::PackagesProposal).to receive("RemoveResolvables")
+        .with("firewall", :package, ["openssh"])
+
+      subject.disable_sshd!
+      expect(subject.enable_sshd).to be(false)
+    end
+
+    it "removes the openssh package for current selection" do
+      expect(Yast::PackagesProposal).to receive("RemoveResolvables")
+        .with("firewall", :package, ["openssh"])
+
+      subject.disable_sshd!
+    end
+  end
+
+  describe "#open_ssh!" do
+    it "sets the ssh port to be opened" do
+      subject.close_ssh!
+      expect(subject.open_ssh).to be(false)
+      subject.open_ssh!
+      expect(subject.open_ssh).to be(true)
```

Comment:
hugh. I don't understand the code well -> most probably will need to go deeply. 

I'm guessing here, but is ```open_ssh``` just a status checker? Shouldn't it be ```opened_ssh?``` instead

argh it is attribute accessor. So, 1) shouldn't be attr_reader enough for ```open_ssh```? Is breaking the encapsulation really needed? 2) this workflow ```close_ssh!``` -> ```open_ssh``` --[port remains closed]--> ```open_ssh!``` --[port finally opened] looks a little bit weird at first view.

Reply:
I could to initializing the attributes, but also considered that it abstract from initialization and you at the end test that both methods are the opposite but as you prefer